daemon/cmd/endpoint.go

```diff
@@ -445,6 +445,16 @@ func (d *Daemon) createEndpoint(ctx context.Context, owner regeneration.Owner, e
 		return d.errorDuringCreation(ep, fmt.Errorf("unable to insert endpoint into manager: %s", err))
 	}
 
+	// We need to make sure DatapathConfiguration.DisableSipVerification value is same as
+	// the value of SourceIPVerification option in the endpoint.
+	if epTemplate.DatapathConfiguration != nil {
+		if epTemplate.DatapathConfiguration.DisableSipVerification {
```

Comment:
I would suggest moving this to https://github.com/cilium/cilium/blob/b76c01164d402b5534fee9398444f0e1fd216c93/pkg/endpoint/endpoint.go#L718 as the `SetDefaultOpts` is called also during endpoint restore (agent restart).
Btw the default is true, so there is no need to have the else statement =>
```
if e.DatapathConfiguration.DisableSipVerification {
	e.Options.SetValidated(option.SourceIPVerification, option.OptionDisabled)
}
```